kong/kong_test.go

```diff
@@ -162,3 +173,40 @@ func TestRunWhenEnterprise(T *testing.T) {
 	v := root["version"].(string)
 	assert.Contains(v, "enterprise")
 }
+
+type HeaderRoundTripper struct {
+	headers []string
+	rt      http.RoundTripper
+}
+
+func NewTestClient(baseURL *string, client *http.Client) (*Client, error) {
+	if value, exists := os.LookupEnv("KONG_ADMIN_TOKEN"); exists {
+		c := &http.Client{}
+		defaultTransport := http.DefaultTransport.(*http.Transport)
+		c.Transport = defaultTransport
+		c.Transport = &HeaderRoundTripper{
+			headers: []string{fmt.Sprintf("kong-admin-token:%v", value)},
+			rt:      defaultTransport,
+		}
+		return NewClient(baseURL, c)
+	}
+	return NewClient(baseURL, client)
+}
+
+// RoundTrip satisfies the RoundTripper interface.
+func (t *HeaderRoundTripper) RoundTrip(req *http.Request) (*http.Response,
```

Comment:
We already implement a HeaderRoundTripper:
https://github.com/hbagdi/go-kong/blob/master/kong/utils.go#L79

Please re-use this one.